ReadersAndroidJetpack/app/src/main/java/com/kontinua/readersandroidjetpack/data/BookmarkRepository.kt

```diff
@@ -0,0 +1,47 @@
+package com.kontinua.readersandroidjetpack.data
+
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringPreferencesKey
+import androidx.datastore.preferences.preferencesDataStore
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import kotlinx.serialization.builtins.MapSerializer
+import kotlinx.serialization.builtins.SetSerializer
+import kotlinx.serialization.builtins.serializer
+import kotlinx.serialization.json.Json
+
+val Context.bookmarkDataStore: DataStore<Preferences> by preferencesDataStore(name = "bookmarks_preferences")
+
+class BookmarkRepository(private val context: Context) {
+
+    private object PreferencesKeys {
+        val BOOKMARK_LOOKUP_JSON = stringPreferencesKey("bookmark_lookup_json")
+    }
+
+    private val mapSerializer = MapSerializer(Int.serializer(), SetSerializer(Int.serializer()))
```

Comment:
How are you handling changes in workbooks over time? Do things behave OK if you bookmark the last page, then the next rev is shorter by a page?

Are the workbook IDs stable over time by design? Or could they be reordered as the curriculum evolves?